packages/aws-cdk-lib/aws-opensearchservice/lib/domain.ts

```diff
@@ -1685,6 +1757,18 @@ export class Domain extends DomainBase implements IDomain, ec2.IConnectable {
             masterUserName: masterUserName,
             masterUserPassword: this.masterUserPassword?.unsafeUnwrap(), // Safe usage
           },
+          samlOptions: samlAuthenticationEnabled ? {
+            enabled: true,
+            idp: {
```

Comment:
How about something like this so we can remove the `!` operator:
```ts
idp: props.fineGrainedAccessControl && props.fineGrainedAccessControl.samlAuthenticationOptions ? {
  entityId: props.fineGrainedAccessControl.samlAuthenticationOptions.idpEntityId,
  metadataContent: props.fineGrainedAccessControl.samlAuthenticationOptions.idpMetadataContent,
} : undefined,
```
What do you think?